packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,220 @@
+
+// eslint-disable-next-line import/no-internal-modules
+import { loadOffline, getPendingOps } from "./offline/offlineTestsUtils.js";
+
+const waitForSummary = async (
+	provider: ITestObjectProvider,
+	container: IContainer,
+	testContainerConfig: ITestContainerConfig,
+	summaryVersion?: string,
+) => {
+	const { summarizer, container: summarizingContainer } = await createSummarizer(
+		provider,
+		container,
+		testContainerConfig,
+		summaryVersion,
+	);
+	await provider.ensureSynchronized();
+	const result = await summarizeNow(summarizer);
+	summarizingContainer.close();
+	return result.summaryVersion;
+};
+
+const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+	getRawConfig: (name: string): ConfigTypes => settings[name],
+});
+
+describeCompat(
+	"Offline tests that wait for a summary",
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const mapId = "map";
+		const { SharedMap } = apis.dds;
+		const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
+		let provider: ITestObjectProvider;
+		let loader: IHostLoader;
+		let container: IContainerExperimental;
+		let url: any;
+		let map1: ISharedMap;
+		let dataStore1: ITestFluidObject;
+		const testContainerConfig: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			registry,
+			loaderProps: {
+				configProvider: configProvider({
+					"Fluid.Container.enableOfflineLoad": true,
+					"Fluid.Sequence.intervalStickinessEnabled": true,
+				}),
+			},
+		};
+		const mainContainerConfig: ITestContainerConfig = {
+			...testContainerConfig,
+			runtimeOptions: {
+				summaryOptions: {
+					summaryConfigOverrides: {
+						state: "disabled",
+					},
+				},
+			},
+		};
+
+		beforeEach("setup", async () => {
+			provider = getTestObjectProvider({ syncSummarizer: true });
+			loader = provider.makeTestLoader(mainContainerConfig);
+			container = await createAndAttachContainer(
+				provider.defaultCodeDetails,
+				loader,
+				provider.driver.createCreateNewRequest(provider.documentId),
+			);
+			provider.updateDocumentId(container.resolvedUrl);
+			url = await container.getAbsoluteUrl("");
+			dataStore1 = (await container.getEntryPoint()) as ITestFluidObject;
+			map1 = await dataStore1.getSharedObject<ISharedMap>(mapId);
+			map1.set("1", "1");
+		});
+
+		it("works with summary while offline", async function () {
+			const summaryVersion = await waitForSummary(provider, container, testContainerConfig);
+			const pendingOps = await getPendingOps(
+				testContainerConfig,
+				provider,
+				false, // Don't send ops from first container instance before closing
+				async (c, d) => {
+					const map: ISharedMap = await d.getSharedObject<ISharedMap>(mapId);
+					map.set("stashed", "stashed");
+				},
+			);
+
+			map1.set("2", "2");
+			await waitForSummary(provider, container, testContainerConfig, summaryVersion);
+			const container2 = await loader.resolve({ url }, pendingOps);
```

Comment:
Is it intentional that container2 doesn't load from the summary that happens before this step?